handler/src/main/java/io/netty/handler/ssl/OpenSslClientContext.java

```diff
@@ -188,51 +198,37 @@ public OpenSslClientContext(File trustCertCollectionFile, TrustManagerFactory tr
                 ClientAuth.NONE);
         boolean success = false;
         try {
-            checkKeyManagerFactory(keyManagerFactory);
             if (key == null && keyCertChain != null || key != null && keyCertChain == null) {
                 throw new IllegalArgumentException(
                         "Either both keyCertChain and key needs to be null or none of them");
             }
             synchronized (OpenSslContext.class) {
-                if (keyCertChain != null && key != null) {
-                    /* Load the certificate file and private key. */
-                    long keyBio = 0;
-                    long keyCertChainBio = 0;
-
-                    try {
-                        keyCertChainBio = toBIO(keyCertChain);
-
-                        keyBio = toBIO(key);
-
-                        if (!SSLContext.setCertificateBio(
-                                ctx, keyCertChainBio, keyBio, keyPassword, SSL.SSL_AIDX_RSA)) {
-                            long error = SSL.getLastErrorNumber();
-                            if (OpenSsl.isError(error)) {
-                                throw new SSLException("failed to set certificate and key: "
-                                                       + SSL.getErrorString(error));
-                            }
+                try {
+                    if (!OpenSsl.supportsKeyManagerFactory()) {
+                        if (keyManagerFactory != null) {
+                            throw new IllegalArgumentException(
+                                    "KeyManagerFactory not supported");
                         }
-                        // We may have more then one cert in the chain so add all of them now. We must NOT skip the
-                        // first cert when client mode.
-                        if (!SSLContext.setCertificateChainBio(ctx, keyCertChainBio, false)) {
-                            long error = SSL.getLastErrorNumber();
-                            if (OpenSsl.isError(error)) {
-                                throw new SSLException(
-                                        "failed to set certificate chain: " + SSL.getErrorString(error));
-                            }
+                        if (keyCertChain != null && key != null) {
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Change this condition so that it does not always evaluate to "true" [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS2583)